src/core/storage/fileio/general_fstream.cpp

```diff
@@ -21,27 +22,33 @@ general_ifstream::general_ifstream(std::string filename)
     // this is the function try block syntax which, together with the member
     // function pointer syntax, is probably the ugliest C++ syntactic element
     // every conceieved.
-    try :general_ifstream_base(filename), opened_filename(filename)
-    {
-    }  catch (const std::exception& e) {
-      log_and_throw_io_failure("Cannot open " + sanitize_url(filename) + " for read. " + e.what());
-    } catch (std::string e) {
-      log_and_throw_io_failure("Cannot open " + sanitize_url(filename) + " for read. " + e);
-    } catch (...) {
-      log_and_throw_io_failure("Cannot open " + sanitize_url(filename));
-    }
-
-
-general_ifstream::general_ifstream(std::string filename, bool gzip_compressed)
-    try :general_ifstream_base(filename, gzip_compressed),
-     opened_filename(filename) {
-     } catch (const std::exception& e) {
-      log_and_throw_io_failure("Cannot open " + sanitize_url(filename) + " for read. " + e.what());
-    } catch (std::string e) {
-      log_and_throw_io_failure("Cannot open " + sanitize_url(filename) + " for read. " + e);
-    } catch (...) {
-      log_and_throw_io_failure("Cannot open " + sanitize_url(filename));
-    }
+    try : general_ifstream_base(filename),
+          opened_filename(filename) {
+  exceptions(std::ios_base::badbit);
```

Comment:
That's why you may need, http://www.cplusplus.com/reference/ios/ios/exceptions/

badbit includes failbit, anyway.